common/src/main/java/com/rln/client/damlClient/RLNDamlClient.java

```diff
@@ -186,6 +189,34 @@ public void archiveBalance(ArchiveBalanceParameters parameters) {
         commandPublisher.onNext(new ClientCommand(event, update, parameters.getProvider(), parameters));
     }
 
+    @Override
+    public void changeBalance(ChangeBalanceParameters parameters) {
+        var event = String.format("Changing balance %s (provider: %s)", parameters.getIban(), parameters.getProvider());
+        if (parameters.getChange().compareTo(BigDecimal.ZERO) == 0) {
+          return;
+        }
+        Update<?> update;
+        if (parameters.getChange().compareTo(BigDecimal.ZERO) < 0) {
+          update = Balance.byKey(new BalanceKey(parameters.getProvider(), parameters.getIban())).exerciseDecrease(parameters.getChange().abs());
+        } else {
+          update = Balance.byKey(new BalanceKey(parameters.getProvider(), parameters.getIban())).exerciseIncrease(parameters.getChange().abs());
+        }
```

Comment:
I would really like to see this extracted to a method, however I don't really have a good idea for a name. 🤷‍♂️